File: datalib/src/main/java/nl/acidcats/tumblrlikes/core/usecases/checktime/CheckTimeUseCase.java
package nl.acidcats.tumblrlikes.core.usecases.checktime;


import rx.Observable;

/**
 * Created on 16/10/2018.
 */
public interface CheckTimeUseCase {

    Observable<Boolean> isTimeToCheck(long currentTimeInMs);

    Observable<Long> resetCheckTime();
}

File: datalib/src/main/kotlin/nl/acidcats/tumblrlikes/core/usecases/checktime/CheckTimeUseCase.kt
package nl.acidcats.tumblrlikes.core.usecases.checktime

import rx.Observable

/**
 * Created on 24/10/2018.
 */
interface CheckTimeUseCase {
    fun isTimeToCheck(currentTimeInMs: Long): Observable<Boolean>

    fun setLastCheckTime(currentTimeInMs: Long): Observable<Long>

    fun resetCheckTime(): Observable<Long>
}